### ISSUE TITLE
Relative reality tabs offset

### ISSUE THREAD

**Issue by modulesio** [webmixedreality/exokit#701]
Reality tabs offsets in the world are controlled by the `xrOffset : XRRigidTransform` parameters of the owning `iframe`.

The user rightly expects that this would be a relative offset from the parent (owning document) but that's not how it currently works. Currently it's just an absolute offset applied to the root context.

The correct way to do with would be to stack-multiply out the transform matrices from root to child, so that reality tabs form a sort of compositor scene graph.

**Comment by shaneharris** [comment]
yes with more playing with this i found that the child scenes seem to work ok if they use XR sessions - i have yet to confirm this for sure though. All the examples in the repo work as expected but if you pull a random three.js/aframe example they seem fixed in their absolute space. 

I will get a repro together for this but i was able to relaibly reproduce by changing the default "exobot.html" url in the realitytabs.html example to say `https://threejs.org/examples/webgl_animation_cloth.html`

**Comment by shaneharris** [comment]
Interesting, it actually does render fine but ignores the offset in that case. It would be amazing to be able to load any old three.js / aframe scene into a tab and explore it. I might spend some time looking to see if XR support could be injected into the scene and then have some support for recent scenes at least. All the three.js examples for instance use the latest THREE.js so the support is there really.

**Comment by modulesio** [comment]
> It would be amazing to be able to load any old three.js / aframe scene into a tab and explore it.

Unless exokit starts to be tied to THREE.js, this sounds logically impossible. Exokit's API is WebGL/WebXR and it doesn't know about how a site is rendering. We could special-case for certain versions of certain frameworks or try to detect the rendering discipline with hacks, but this isn't doable in the general case.

See: Emukit's N64VR/PSXVR

**Comment by modulesio** [comment]
That said, I think there's room for an XRifier script for THREE.js examples. Actually I think @ChrisEddy made some awesome headway there.

**Comment by shaneharris** [comment]
oh yeh indeed sorry i meant to say i would come up with this separately. Oh @ChrisEddy id be interested to look at that! I have some ideas of grabbing the renderer and fiddling the animation loop essentially but id be interested to see what you have got.

**Comment by ChrisEddy** [comment]
Hey so awhile back I forked three.js-examples and "VRified" some of them. There is one 20 line code snippet that works on all the examples usually. But where you have to think, is you usually have to go through and adjust all the scales of objects and make sure no camera code is active and messing with the VR camera. I'll post my fork.

**Comment by ChrisEddy** [comment]
https://github.com/ChrisEddy/exokit-three.js-example-library it's a bit out of date, and was still using rawgit to share lol, but if you look at files that I commited to, you can get the idea.

**Comment by ChrisEddy** [comment]
https://github.com/ChrisEddy/exokit-three.js-example-library/blob/master/three.js-dev/examples/webgl_geometry_hierarchy2.html this one works the best in Exokit and looks cool, the VRified code snippet is at the bottom of code

**Comment by shaneharris** [comment]
yeh they still work for a few years. there are a few alternatives now. Im using `raw.gitack.com`. Cheers for the info and links ill get stuck into that!
